email_template_qweb/models/mail_template.py

```diff
@@ -40,3 +42,36 @@ def generate_email(self, res_ids, fields=None):
                         result[res_id]['body_html']
                     )
         return multi_mode and result or result[res_ids[0]]
+
+    def _qweb_render_context(self, record):
+        res = {}
+        if record._name == 'mail.message':
+            partner_model = self.env['res.partner']
+            # these variables are usually loaded when the notification is sent
+            # but there are some key values that that are nice to have
+            # if you want to pre-render your template in preview mode.
+            # This is particularly useful if you use the template
+            # from the mail composer, whereas the rendering is done on load.
+            # Also, you don't need to access them from `ctx`.
+            res = partner_model._notify_prepare_template_context(record)
+        res.update({
+            'object': record,
+            'email_template': self,
+            # Same as for Jinja rendering,
+            # taken from `mail_template.render_template`.
+            # These ease porting of old Jinja templates to qweb ones.
+            'format_date':
+                lambda date, format=False,
+                context=self._context: format_date(self.env, date, format),
```

Comment:
I don't see why we need jinja compatibility. If you want to format a date or money, you just use the proper widget in qweb

Reply:
@hbrunn as I mentioned: it eases reusing jinja templates. It makes easier for end users to customize templates. 

I could make them optional if you want or split to another module but seems over complicated for such a small thing.

Regarding "the whole point": I disagree, qweb is not only for using widgets and such. The key point about qweb is that you can reuse templates and its syntax is much more readable.

This is just a tiny layer of compatibility between them.

Reply:
I could remove the formatters if we agree is "too much" :)

Reply:
no, keep it the way you have it. It doesn't hurt my qweb purism and helps others, so it's a good thing to do, thank you.

But https://codecov.io/gh/OCA/social/compare/b6dc1918b4aa8670a303edff99fcedeb3391250c...3efbe62cb6dea0b91e87be7391b750cc393a2553/diff#D2-44 needs to be addressed